plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -494,6 +502,21 @@ private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNo
         return new DeltaLakeColumnMetadata(columnMetadata, fieldName, fieldId, physicalName, physicalColumnType);
     }
 
+    private static void verifyTypeChange(JsonNode typeChange)
+    {
+        String fromType = typeChange.get("fromType").asText();
+        String toType = typeChange.get("toType").asText();
+
+        switch ("%s -> %s".formatted(fromType, toType)) {
+            case "byte -> short":
+            case "short -> integer":
+                return;
```

Comment:
I can promote byte to  short and then short to integer.
can i promote byte to integer?

Reply:
We can promote byte to integer. Added another condition and the product test.